StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1131UseReadableConditions.cs

```diff
@@ -86,7 +86,11 @@ private static bool IsLiteral(ExpressionSyntax expression, SemanticModel semanti
             switch (symbol)
             {
             case IFieldSymbol fieldSymbol when fieldSymbol.IsStatic && fieldSymbol.IsReadOnly:
-            case IMethodSymbol:
+                return true;
+
+            // NOTE: Without the when clause, this would also treat e.g unary or binary expressions as literals,
+            // since GetSymbolInfo returns the operator method in those cases.
+            case IMethodSymbol when expression is IdentifierNameSyntax:
```

Comment:
📝 I think this needs to be `NameSyntax` instead. Also the tests should include some examples involving qualified methods (e.g. `TestClass.Method1`) and generic methods (e.g. `Method3<int>`, `Enum.TryParse<DateTimeKind>`)

Reply:
Updated. Did not add test for Enum.TryParse since I couldn't get it to compile (due to multiple overloads) in time.

I have to leave now. Feel free to revert commit dfd5ed2e454d71a563dac50ca7bed57a37d31e9f instead if you would like. I can redo that one later on if you see more problems with this and you would like to make a new beta quickly.